Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -397,36 +397,100 @@ private class AttributeRemover: SyntaxRewriter {
     var filteredAttributes: [AttributeListSyntax.Element] = []
     for case .attribute(let attribute) in node {
       if attributesToRemove.contains(attribute) {
-        var leadingTrivia = node.leadingTrivia
-        if let lastNewline = leadingTrivia.pieces.lastIndex(where: { $0.isNewline }),
-          leadingTrivia.pieces[lastNewline...].allSatisfy(\.isWhitespace),
-          node.trailingTrivia.isEmpty,
-          node.nextToken(viewMode: .sourceAccurate)?.leadingTrivia.first?.isNewline ?? false
+        var leadingTrivia = attribute.leadingTrivia
+        if let lastNewline = leadingTrivia.pieces.lastIndex(where: \.isNewline),
+          attribute.trailingTrivia.isEmpty,
```

Comment:
Since you removed the `leadingTrivia.pieces[lastNewline...].allSatisfy(\.isWhitespace)` check, how do we handle the following now? I think we’ll remove the block comment but we shouldn’t.

```swift
@State
/*doc comment*/@Test
var x: Int
```